src/mito2/src/sst/index/fulltext_index/creator.rs

```diff
@@ -0,0 +1,293 @@
+    pub fn memory_usage(&self) -> usize {
+        self.creators.values().map(|c| c.inner.memory_usage()).sum()
+    }
+
+    /// Returns IDs of columns that the creator is responsible for.
+    pub fn column_ids(&self) -> impl Iterator<Item = ColumnId> + '_ {
+        self.creators.keys().copied()
+    }
+
+    pub(crate) fn is_empty(&self) -> bool {
+        self.creators.is_empty()
+    }
+}
+
+impl SstIndexCreator {
+    async fn do_update(&mut self, batch: &Batch) -> Result<()> {
+        let mut guard = self.stats.record_update();
+        guard.inc_row_count(batch.num_rows());
+
+        for creator in self.creators.values_mut() {
+            creator.update(batch).await?;
+        }
+
+        Ok(())
+    }
+
+    async fn do_finish(&mut self, puffin_writer: &mut SstPuffinWriter) -> Result<()> {
+        let mut guard = self.stats.record_finish();
+
+        let mut written_bytes = 0;
+        for creator in self.creators.values_mut() {
+            written_bytes += creator.finish(puffin_writer).await?;
+        }
+
+        guard.inc_byte_count(written_bytes);
+        Ok(())
+    }
+
+    async fn do_abort(&mut self) -> Result<()> {
+        let _guard = self.stats.record_cleanup();
+
+        self.aborted = true;
+
+        for (_, mut creator) in self.creators.drain() {
+            creator.abort().await?;
+        }
+
+        Ok(())
+    }
+}
+
+/// `SingleCreator` is a creator for a single column.
+struct SingleCreator {
+    /// Column ID.
+    column_id: ColumnId,
+    /// Inner creator.
+    inner: Box<dyn FulltextIndexCreator>,
+    /// Intermediate path where the index is written to.
+    intm_path: PathBuf,
+    /// Whether the index should be compressed.
+    compress: bool,
+}
+
+impl SingleCreator {
+    async fn update(&mut self, batch: &Batch) -> Result<()> {
+        let text_column = batch
+            .fields()
+            .iter()
+            .find(|c| c.column_id == self.column_id);
+        match text_column {
+            Some(column) => {
+                let data = column
+                    .data
+                    .cast(&ConcreteDataType::string_datatype())
+                    .context(CastVectorSnafu {
+                        from: column.data.data_type(),
+                        to: ConcreteDataType::string_datatype(),
+                    })?;
+
+                for i in 0..batch.num_rows() {
+                    let data = data.get_ref(i);
+                    let text = data
+                        .as_string()
+                        .context(FieldTypeMismatchSnafu)?
+                        .unwrap_or_default();
+                    self.inner
+                        .push_text(text)
+                        .await
+                        .context(FulltextPushTextSnafu)?;
+                }
+            }
+            _ => {
+                for _ in 0..batch.num_rows() {
+                    self.inner
+                        .push_text("")
```

Comment:
Not quite sure, in such a case,do we still need to index the empty string?

Reply:
If the `text` column in a certain batch is all `None`, we still need to `push("")` to ensure the row numbers in the index match.

Reply:
@zhongzc Got it, could you add a comment for it? Other people may have the same question.